test/auth-controller-test.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+process.env.APP_SECRET = 'something123';
+process.env.MONGODB_URI = 'mongodb://localhost/rambletest';
+
+const expect = require('chai').expect;
+
+const server = require('../server');
+const port = process.env.PORT || 3000;
+
+const authController = require('../controller/auth-controller');
+const userController = require('../controller/user-controller');
+
+describe('testing auth-controller functions', function(){
+  before( (done) => {
+    if(!server.isRunning){
+      server.listen(port, () => {
+        server.isRunning = true;
+        console.log('server started booyahhackerhackeretcetc', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after( (done) => {
+    if(server.isRunning){
+      server.close( () => {
+        server.isRunning = false;
+        console.log('server be down');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('test signup() function', function(){
+    after( (done) => {
+      userController.removeAllUsers()
+      .then(() => done())
+      .catch(done);
+    });
+
+    it('should return a new user', (done) => {
+      authController.signup({username: 'testuser', password: 'testpassword'})
```

Comment:
Great job mocking your data using the controllers instead of superagent request!!!